### ISSUE TITLE
[Feature request] Simulation API is hard to use for certain use cases due to required params

### ISSUE THREAD

**Issue by movekevin** [aptos-labs/aptos-core#6862]
# 🐛 Feature request

Currently in order to successfully execute a simulation of a tx, user is required to provide:
1. An invalid signature to prevent replay.
2. A correct public key is provided that matches the sender account's auth key
3. A correct sequence number

This is fine if the use case is to simulate before sending the actual transaction with the same exact params. In many cases, a DApp just wants to run the simulation to understand the outcome or to test specific transactions for accounts that they don't have the public keys for.

## Proposal
We relax the requirements for (2) and (3) or skip transaction prologue altogether. For specific checks such as account having enough to pay gas, this is already covered in tx epilogue.

**Comment by e99243506bigplay** [comment]
I would like also require a batch transactions simulation.

**Comment by alnoki** [comment]
@davidiw Per our recent in-person discussion the current simulator design prohibits simulation of transactions from a multisig v2 account, which does not have a public key.

@movekevin the checks on account sequence number and gas are also problematic for multisig v2 accounts:

* The `Account.sequence_number` of a multisig v2 account should always be 0, since it is only the `MultisigAccount.sequence_number` that gets incremented.
* In most cases a multisig v2 account will not have any gas available, because the gas fee payer is typically an owner who executes transactions via a `MultisigTransactionPayload`

Additionally, the requirement to include a well-formed but invalid signature is confusing and cumbersome.

Hence per https://github.com/aptos-labs/aptos-core/issues/6862#issue-1607104987 , I suggest removing requirements 1, 2, and 3.

**Comment by alnoki** [comment]
>  It's not always correct that the sequence number can be ignored as we do allow an existing account (might or might not have an multi-ed25519 key) to migrate itself to become a multisig account

@movekevin thanks for pointing out this possibility that I overlooked

> The current gas and nonce checks actually apply to the owner who sends the transaction and are not levied against the multisig account.

I think the core issue here is simulating an arbitrary transaction that requires the `&signer` of a `MultisigAccount`, rather than of an individual owner: right now for such a transaction to be simulated it would first have to pass proposal flow and have enough votes to execute, which could lead to a situation of a pending transaction that no one actually wants to execute, but rather was voted on as yes just because the simulation was of interest.

Basically for a multisig there is no way to simulate the outcome of an executed multisig transaction (where the `&signer` of a `MultisigAccount` is generated) before proposal flow, even though proposal flow would ideally take place only after a simulation was performed.

**Comment by movekevin** [comment]
Actually on the contrary, if you recall https://github.com/aptos-labs/aptos-core/issues/8304, currently simulation of multisig txs (effectively acting as the multisig signer) only works for proposals that have NOT been created. It doesn't support simulating executing pending multisig txs (those that have already been created and might have gone through proposal flow). Once that ticket is closed, we'll have supported both - simulating txs that have or have not been created. 

Let's also take any multisig-specific discussion over to the other ticket so we can keep this one focused on improving the simulation API for non-multisig txs

**Comment by gregnazario** [comment]
Similarly let's check up on this @junkil-park
